core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java

```diff
@@ -2436,56 +2409,44 @@ public Integer call() throws IOException {
                 return doAbortX(regionName, transactionState.getTransactionId(), participantNum, location.isTableRecodedDropped());
               }
             });
-/*
-          } catch (Exception e) {
-            LOG.error("exception in abort: " + e);
-          }
-*/
-            /*
-            } catch (UnknownTransactionException e) {
-        LOG.error("exception in abort: " + e);
-                LOG.info("Got unknown transaction exception during abort. Transaction: ["
-                        + transactionState.getTransactionId() + "], region: ["
-                        + location.getRegionInfo().getRegionNameAsString() + "]. Ignoring.");
-            } catch (NotServingRegionException e) {
-                LOG.info("Got NSRE during abort. Transaction: [" + transactionState.getTransactionId() + "], region: ["
-                        + location.getRegionInfo().getRegionNameAsString() + "]. Ignoring.");
-            }
-            */
         }
 
         // all requests sent at this point, can record the count
         transactionState.completeSendInvoke(loopCount);
     }
+         
+        CommitUnsuccessfulException savedCue = null;
 
         //if DDL is involved with this transaction, need to unwind it.
         if(transactionState.hasDDLTx())
         {
 
             //First wait for abort requests sent to all regions is received back.
             //This TM thread gets SUSPENDED until all abort threads complete!!!
-            try{
+            boolean loopExit = false;
+            do
+            {
+              try {
                 transactionState.completeRequest();
-            }
-            catch(Exception e){
-                LOG.error("Exception in abort() completeRequest. txID: " + transactionState.getTransactionId() + "Exception: " + e);
-                //return; //Do not return here. This thread should continue servicing DDL operations.
-            }
-
-            try{
-                abortDDL(transactionState);
-            }
-            catch(Exception e){
-                LOG.error("FATAL Exception calling abortDDL for transaction: " + transactionState.getTransactionId() + "Exception: "  + e);
-                throw new UnsuccessfulDDLException(e);
-            }
+                loopExit = true; 
+              } 
+              catch (InterruptedException ie) {}
+              catch (CommitUnsuccessfulException cue) {
+                 loopExit = true;
+                 LOG.error("Exception at the time of aborting DDL transaction", cue); 
```

Comment:
AbortDDL is not started yet here. The error message needs to get updated something like "CommitUnSuccessful Exception received from one or more of abort threads before servicing abortDDL"